test/sqllogictest/transform/column_knowledge.slt

```diff
@@ -650,21 +649,19 @@ SELECT f1, f2, f1 IS NOT NULL, f2 IS NULL FROM c0;
 ----
 Explained Query:
   Return // { arity: 4, types: "(integer?, integer?, boolean, boolean)" }
-    Map (true, false) // { arity: 4, types: "(integer?, integer?, boolean, boolean)" }
+    Map (true, (#1) IS NULL) // { arity: 4, types: "(integer?, integer?, boolean, boolean)" }
       Get l0 // { arity: 2, types: "(integer?, integer?)" }
   With Mutually Recursive
     cte l0 =
       Distinct group_by=[#0, #1] // { arity: 2, types: "(integer?, integer?)" }
         Union // { arity: 2, types: "(integer?, integer?)" }
           Get materialize.public.t1 // { arity: 2, types: "(integer, integer?)" }
-          Get l0 // { arity: 2, types: "(integer?, integer?)" }
+          Filter (#1) IS NOT NULL // { arity: 2, types: "(integer?, integer)" }
+            Get l0 // { arity: 2, types: "(integer?, integer?)" }
 
 EOF
 
 # Multiple bindings, value knowledge
-#
-# This also illustrates a missed opportunity here, because if we are a bit
```

Comment:
Changes between `v1` and `v2` are because we can now handle the constant propagation of `l1.#0 = 2`.